### ISSUE TITLE
Could Istio maybe be installed through official channels for that project?

### ISSUE THREAD

**Issue by krancour** [knative/docs#299]
Is there a reason we "repackage" Istio in [this manifest](https://storage.googleapis.com/knative-releases/serving/latest/istio.yaml)?

imo, unless a very specific Istio configuration is required, it should be installable via the official channels for that project instead-- e.g. Istio's Helm chart.

Even if Istio _does_ require a very specific configuration, I would suggest that documenting that is preferable to re-packaging Istio, which leaves users to speculate how that Istio installation differs from the norm.

**Comment by sslavic** [comment]
From last week's TGIK session:
- one can find two `PATCH`es labeled in https://storage.googleapis.com/knative-releases/serving/latest/istio.yaml
  - first patch adds istio-system namespace, to minimize number of installation steps because compared to Helm chart it's possible with single plain yaml to install namespace and reference it too
  - second patch is for https://github.com/knative/serving/issues/1370
- Knative's Istio plain yaml for the most part (everything but two patches) is generated from upstream Istio Helm chart but Helm is not used for deployment e.g. Tiller is hard to install in a secure way

**Comment by tcnghia** [comment]
Agreed.  Being pluggable means we should allow users to customize their own installation of Istio, and if a customer already has their own way of installing Istio they don't need to reinstall based on what we provide.

That said, we can see the Istio we provide as an example setting for a quick start experience, so our users don't need to first install Helm and read many Istio getting started doc to figure out what options need to be passed to Helm.

In the `repackaging` we documented all the Helm options we used to generate the YAML, I think we could bring similar docs in the `knative/docs` repo (as you suggested).  We should fix knative/serving#1370 so the entire thing could be done within a single Helm command.

**Comment by krancour** [comment]
> Tiller is hard to install in a secure way

Absolutely no disagreement from me on that point.

I'm more worried about drift, and as you've mentioned, pluggability. byo-Istio should not only be possible (it is-- I've done it), but also considered a likely scenario. Istio's about to go 1.0.0 and Knative is still very young. It won't be long before we start encountering clusters in the wild that already have Istio installed to their liking.

Maybe after #1370 is fixed, could the possibility of drift be mitigated for now (while avoiding Tiller security issues) by doing this instead of repackaging?

```
$ kubectl create ns istio-system
$ helm template install/kubernetes/helm/istio --name istio --namespace istio-system --set | kubectl apply -f -
```

**Comment by krancour** [comment]
Also see https://github.com/istio/istio/issues/7505

**Comment by krancour** [comment]
Although there could be a gotcha here in that `helm template` seems to lack the `--version` flag that `helm install`. That make installation of a specific version of the chart difficult to document.

I'll open a related feature request against Helm.

**Comment by samodell** [comment]
@tcnghia  Can you give an update on whether or not we have plans to install Istio from an "official" source instead of providing our own .yaml files for users to install from?

**Comment by tcnghia** [comment]
Yes, we should add documentation about using users' own Istio.  We could probably document the point of our patches - may be pointing it to issues like https://github.com/istio/istio/issues/12183 to highlight that our patches are only to improve Istio stability.

The Istio yaml we provided currently is already obtained through the official channels of Istio (through this script https://github.com/knative/serving/blob/master/third_party/istio-1.0.6/download-istio.sh) so the bulk of this work is adding more docs.

**Comment by samodell** [comment]
Thanks @tcnghia ! Let me know if you/anyone has time to draft a few paragraphs on these topics. Docs team can then polish and figure out where to put the info.

**Comment by samodell** [comment]
Related: https://github.com/knative/serving/issues/3670

**Comment by mattmoor** [comment]
Can we prioritize this for 0.6?

**Comment by RichieEscarez** [comment]
Related: #1148

**Comment by RichieEscarez** [comment]
Istio install docs published for 0.6: https://knative.dev/docs/install/installing-istio/
